lib/taste_tester/client.rb

```diff
@@ -136,53 +136,48 @@ def populate(stream, writer, path, destination)
-          # level below the directory included here. Strip off the first part
-          # of the path. e.g. `fb_cookbook/spec/foo.rb` would be `spec/foo.rb`
-          # which matches `spec/*`.
-          relative_name_minus_first = relative_name.split(
-            File::SEPARATOR,
-          )[1..-1].join(File::SEPARATOR)
-          next if chefignores.ignored?(relative_name) ||
-            chefignores.ignored?(relative_name_minus_first)
-          name = File.join(destination, relative_name)
-          if File.directory?(p)
-            # skip it. This also handles symlinks to directories which aren't
-            # useful either.
-          elsif File.symlink?(p)
-            # tar handling of filenames > 100 characters gets complex. We'd use
-            # split_name from Minitar, but it's a private method. It's
-            # reasonable to assume that all symlink names in the bundle are
-            # less than 100 characters long. Long term, the version of minitar
-            # in chefdk should be upgraded.
-            fail 'Add support for long symlink paths' if name.size > 100
-            # The version of Minitar included in chefdk does not support
-            # symlinks directly. Therefore we use direct writes to the
-            # underlying stream to reproduce the symlinks
-            symlink = {
-              :name => name,
-              :mode => 0644,
-              :typeflag => '2',
-              :size => 0,
-              :linkname => File.readlink(p),
-              :prefix => '',
-            }
-            stream.write(Minitar::PosixHeader.new(symlink))
-          else
-            File.open(p, 'rb') do |r|
-              writer.add_file_simple(
-                name, :mode => 0644, :size => File.size(r)
-              ) do |d, _opts|
-                IO.copy_stream(r, d)
+        look_at = ['']
+        while (prefix = look_at.pop)
+          Dir.glob("#{prefix}**/*", File::FNM_DOTMATCH) do |p|
+            minus_first = p.split(
+              File::SEPARATOR,
+            )[1..-1].join(File::SEPARATOR)
+            next if chefignores.ignored?(p) ||
+              chefignores.ignored?(minus_first)
+            name = File.join(destination, p)
+            if File.directory?(p)
+              # we don't store directories in the tar, but we do want to follow
+              # top level symlinked directories as they are used to share
+              # cookbooks between codebases.
+              if minus_first == '' && File.symlink?(p)
+                look_at.push("#{p}#{File::SEPARATOR}")
+              end
+            elsif File.symlink?(p)
+              # tar handling of filenames > 100 characters gets complex. We'd
+              # use split_name from Minitar, but it's a private method. It's
+              # reasonable to assume that all symlink names in the bundle are
+              # less than 100 characters long. Long term, the version of minitar
+              # in chefdk should be upgraded.
```

Comment:
Can you file an issue upstream for this and reference it in the comment?

Reply:
Well chefdk is deprecated. I imagine there's vastly newer minitar in Chef Workstation. @NaomiReeves can probably tell you if it's worth filing a bug or if upgrades-coming-soon-to-a-devserver-near-you will cover it.

Reply:
It's been a while since I looked at this - surely the better path is to explicitly depend on the minimum version of gemspec that fixes this. We can have multiple versions of the gem installed, right?

Reply:
This is an issue in this program first and foremost, so I've captured it here: #139. Can we agree to fix this as a separate issue and get this fix in?